docs/sources/mimir/get-started/about-grafana-mimir-architecture/about-ingest-storage-architecture/index.md

```diff
@@ -0,0 +1,163 @@
+queries that ingesters serve may return stale data and don't provide a read-after-write guarantee.
+
+During normal operations, the delay in ingesting data in ingesters should be below 1 second. This is also known as the end-to-end latency of ingestion.
+
+While this latency is usually insignificant for interactive queries, it can challenge applications that need queries to observe all previous writes.
+For example, the Mimir [ruler](../../../references/architecture/components/ruler) needs samples from earlier rules in a rule group to be available when evaluating later rules in the same group.
+
+To preserve the read-after-write guarantee, clients can add the `X-Read-Consistency: strong` HTTP header to queries. When a query includes this header, the query-frontend fetches the latest offsets from Kafka for all active partitions and propagates these offsets to ingesters. Each ingester waits until it has consumed up to the specified offset for its partition before running the query. This ensures the query observes all samples written to Kafka before the query-frontend received the query, providing strong read consistency at the cost of increased query latency.
+
+By default, the ingester waits up to 20 seconds for the record to be consumed before rejecting the strong read consistency query.
+
+## Kafka
+
+Kafka is a central part of ingest storage architecture. It's responsible for replicating and durably persisting incoming data
+and for making this data available to ingesters.
+
+The use of the Kafka protocol is limited in Grafana Mimir. This allows you to use many Kafka-compatible solutions.
+
+### How Mimir uses Kafka
+
+On the write path, the distributor requires the Kafka cluster to be available for push requests to succeed.
+The distributor issues direct Produce API calls to the Kafka brokers and doesn't produce in transactions or need support for idempotent writes.
+
+Both the ingester and the distributor use the Metadata API to discover the leaders for a topic-partition.
+
+The Mimir ingester uses Kafka consumer groups to persist how far an ingester has consumed from its assigned partition. Each ingester maintains its own consumer group. Each Kafka record is consumed exactly once by each ingester zone and records aren't replayed after being consumed.
+
+The following exceptions exist:
+
+- Abrupt termination: When an ingester terminates abruptly without persisting its offset in the consumer group, the ingester resumes from where the consumer group offset points.
+- Manual configuration: You can configure ingesters to start consuming from the earliest offset, the latest offset, or a specific timestamp of the Kafka topic-partition. In these cases, the ingester uses the ListOffsets API to discover the target offset.
```

Comment:
do we want to give hints on how to configure that? the two flags are `ingest-storage.kafka.consume-from-position-at-startup` and `ingest-storage.kafka.consume-from-timestamp-at-startup`